apps/epic-web/src/inngest/functions/sanity/product.ts

```diff
@@ -0,0 +1,114 @@
+import {getSdk} from '@skillrecordings/database'
+import {inngest} from 'inngest/inngest.server'
+import {updateOrCreateSanityProduct} from 'lib/products'
+import {SANITY_WEBHOOK_EVENT} from 'inngest/events/sanity'
+import groq from 'groq'
+import {createClient} from '@sanity/client'
+
+export const sanityClient = createClient({
+  projectId: process.env.NEXT_PUBLIC_SANITY_PROJECT_ID,
+  dataset: process.env.NEXT_PUBLIC_SANITY_DATASET_ID,
+  useCdn: false,
+  apiVersion: process.env.NEXT_PUBLIC_SANITY_API_VERSION,
+  token: process.env.SANITY_API_TOKEN,
+  perspective: 'published',
+})
+
+async function getProduct(id: string): Promise<any> {
+  const product = await sanityClient.fetch(
+    groq`*[_type == "product" && _id == $id][0] {
+          _id,
+          _type,
+          _updatedAt,
+          _createdAt,
+          productId,
+          title,
+          image,
+          state,
+          "slug": slug.current,
+          body,
+          upgradableTo[]->{
+            ...,
+          },
+          modules[]->{
+            ...,
+          }
+    }`,
+    {id},
+  )
+  return product
+}
+
+export const sanityProductUpdated = inngest.createFunction(
+  {id: `product-update`, name: 'Update Product in Database'},
+  {
+    event: SANITY_WEBHOOK_EVENT,
+    if: 'event.data.event == "product.update"',
```

Comment:
instead of individual events we can filter events like this, makes config boilerplate simpler overall